docs/topics/db/aggregation.txt

```diff
@@ -425,6 +425,25 @@ publisher's that have at least one book with a rating exceeding 3.0. The second
 query asks for the average of a publisher's book's ratings for only those
 ratings exceeding 3.0.
 
+Using multivalued filtering against a previously annotated aggregation might
+still yield duplicate results.
+
+Given:
+
+* Publisher A has two books with each having 100 pages.
+
+Here's an example demonstrating this behavior::
+
+    >>> a = Publisher.objects.annotate(total_pages(Sum("books__pages"))
+    >>> a.total_pages
+    200
+    >>> b = Publisher.objects.annotate(Sum("books__pages")).filter(books__in=Books.objects.all())
+    >>> b.total_pages
+    400
+
+The expected result in both queries should be 200, but subsequent filter or
+additional aggregations can alter the results of the previous one.
```

Comment:
Not sure how to word it properly but the behaviour here is the same as the one documented in [the _Spanning multi-valued relationships_ section](https://docs.djangoproject.com/en/4.1/topics/db/queries/#spanning-multi-valued-relationships).

Maybe we should rephrase the docs to mention that the peculiarities are the same; `filter` calls won't reuse multi-valued JOINs generated by previous `annotate`or `filter` calls. If you want to prevent issues you should filter after annotations or use `Aggregate.filter`.

Reply:
So should I just add that comment about the peculiarities where you marked or do you recommend making changes elsewhere? 

I think mentioning it here will be appropriate and removing the line which sounds similar to the documentation you have linked.